db/collection.go

```diff
@@ -1240,11 +1230,11 @@ func (c *collection) GetAllDocIDs(
 	ctx context.Context,
 	identity immutable.Option[string],
 ) (<-chan client.DocIDResult, error) {
-	txn, err := c.getTxn(ctx, true)
+	ctx, err := ensureContextTxn(ctx, c.db, true)
 	if err != nil {
 		return nil, err
 	}
-
+	txn := mustGetContextTxn(ctx)
```

Comment:
thought: It's a little weird that whe need to extract the `txn` from the context and the next function call passes the context and the transaction.

question: Why not leave it to the function that needs the `txn` to get it from the context?

Reply:
Someone had a preference for this in the standup discussion. If a function requires a transaction it should remain as a parameter and not come from context.